tests/algorithms/test_stochastic_depth.py

```diff
@@ -1,254 +1,252 @@
-    dummy_algorithm = dummy_hparams.initialize_object()
-    old_drop_rates = []
-    get_drop_rate_list(dummy_state.model, drop_rates=old_drop_rates)
-    dummy_state.timer._batch._value = step
-    dummy_algorithm.apply(Event.BATCH_START, dummy_state, noop_dummy_logger)
-    new_drop_rates = []
-    get_drop_rate_list(dummy_state.model, drop_rates=new_drop_rates)
-
-    drop_warmup_iters = dummy_state.steps_per_epoch * dummy_state.max_epochs * dummy_algorithm.drop_warmup
-    assert torch.all(torch.tensor(new_drop_rates) == ((step / drop_warmup_iters) * torch.tensor(old_drop_rates)))
-
-
-@pytest.mark.parametrize("stochastic_method", ['nonsense'])
-def test_invalid_method_name(stochastic_method: str, target_layer_name: str):
-    with pytest.raises(ValueError):
-        sd_hparams = StochasticDepthHparams(stochastic_method=stochastic_method, target_layer_name=target_layer_name)
-        sd_hparams.validate()
-
-
-@pytest.mark.parametrize("target_layer_name", ['nonsense_pt2'])
-def test_invalid_layer_name(stochastic_method: str, target_layer_name: str):
-    with pytest.raises(ValueError):
-        sd_hparams = StochasticDepthHparams(stochastic_method=stochastic_method, target_layer_name=target_layer_name)
-        sd_hparams.validate()
-
-
-@pytest.mark.parametrize("drop_rate", [-0.5, 1.7])
-def test_invalid_drop_rate(stochastic_method: str, target_layer_name: str, drop_rate: float):
-    with pytest.raises(ValueError):
-        sd_hparams = StochasticDepthHparams(stochastic_method=stochastic_method,
-                                            target_layer_name=target_layer_name,
-                                            drop_rate=drop_rate)
-        sd_hparams.validate()
-
-
-@pytest.mark.parametrize("drop_distribution", ['nonsense_pt3'])
-def test_invalid_drop_distribution(stochastic_method: str, target_layer_name: str, drop_distribution: str):
-    with pytest.raises(ValueError):
-        sd_hparams = StochasticDepthHparams(stochastic_method=stochastic_method,
-                                            target_layer_name=target_layer_name,
-                                            drop_distribution=drop_distribution)
-        sd_hparams.validate()
-
-
-@pytest.mark.parametrize("drop_warmup", [-0.5, 1.7])
-def test_invalid_drop_warmup(stochastic_method: str, target_layer_name: str, drop_warmup: float):
-    with pytest.raises(ValueError):
-        sd_hparams = StochasticDepthHparams(stochastic_method,
-                                            target_layer_name=target_layer_name,
-                                            drop_warmup=drop_warmup)
-        sd_hparams.validate()
+
+def test_stochastic_depth_hparams(stochastic_method: str, target_layer_name: str):
+    hparams = StochasticDepthHparams(
+        stochastic_method=stochastic_method,
+        target_layer_name=target_layer_name,
+    )
+    algorithm = hparams.initialize_object()
+    assert isinstance(algorithm, StochasticDepth)
+
+
+class TestSampleBernoulli:
```

Comment:
Stochastic Depths tests are now sorted and grouped by the function they test, and the fixtures are minimally scoped to the class if possible.